### ISSUE TITLE
GALNetwork does not learn: chance accuracy on separable data

### ISSUE THREAD

**Issue by cgrtml** [cgrtml/neural-trees#38]
On three well separated blobs, `GALNetwork` reaches **0.333** training accuracy, which is chance for three classes:

```python
from sklearn.datasets import make_blobs
from neural_trees import GALNetwork
X, y = make_blobs(n_samples=300, centers=3, cluster_std=0.1, random_state=0)
GALNetwork(max_epochs=100, random_state=0).fit(X, y).score(X, y)  # 0.333
```

This is why `GALNetwork` fails `check_classifiers_train` in
`sklearn.utils.estimator_checks.check_estimator` (52/55; every other estimator except HMoE is at 55/55).

The likely cause is in `fit`: each epoch takes exactly **one** full-batch SGD step at `lr=0.01`, so `max_epochs=100` means 100 gradient steps total. Growth and pruning decisions are then made on a network that has barely moved from its initialization.

What would close this:

- Train on mini-batches, or take multiple steps per epoch, so the growth criterion sees a network that has actually converged between checks
- Keep the growth and pruning semantics of Alpaydin (1994) intact; this is about the optimizer loop, not the algorithm
- Add a test asserting better than 0.9 accuracy on the blobs above
- Report before and after accuracy on Iris and Wine in the PR

**Difficulty:** Medium
**Good for:** Someone comfortable with PyTorch training loops.
**Scope:** Half a day.
